lib/ddtrace/contrib/aws/patcher.rb

```diff
@@ -0,0 +1,61 @@
+module Datadog
+  module Contrib
+    module Aws
+      COMPATIBLE_WITH = Gem::Version.new('2.0.0')
+      SERVICE = 'aws'.freeze
+      AGENT = 'aws-sdk-ruby'.freeze
+      RESOURCE = 'aws.command'.freeze
+
+      # Responsible for hooking the instrumentation into aws-sdk
+      module Patcher
+        @patched = false
+
+        class << self
+          def patch
+            return @patched if patched? || !compatible?
+
+            require 'ddtrace/ext/app_types'
+            require 'ddtrace/contrib/aws/parsed_context'
+            require 'ddtrace/contrib/aws/instrumentation'
+
+            add_pin
+            add_plugin(Seahorse::Client::Base, *loaded_constants)
+
+            @patched = true
+          rescue => e
+            Datadog::Tracer.log.error("Unable to apply AWS integration: #{e}")
+            @patched
+          end
+
+          def patched?
+            @patched
+          end
+
+          private
+
+          def compatible?
+            return unless defined?(::Aws::VERSION)
+
+            Gem::Version.new(::Aws::VERSION) >= COMPATIBLE_WITH
+          end
+
+          def add_pin
+            Pin.new(SERVICE, app_type: Ext::AppTypes::WEB).tap do |pin|
+              pin.onto(::Aws)
```

Comment:
assuming that each application uses only one initialized AWS client, setting the `Pin` in the module `::Aws` is fine.

Reply:
Actually each service builds its own client dynamically, the first time the service constant (e.g. `S3`) is referenced. If you think having a separated pin would be desirable, I could work on that, but I'll probably have to do some nasty hack around the `inherited` hook of the base class or something like that. That would make the contribution a bit more brittle, because we wouldn't be using their plugin interface. Whatever you think it's best!

Reply:
I think that for a user AWS is the service. I prefer a simple approach and then extend it if it's needed. Especially if we should do a nasty hack, so good to me that way 👍